### ISSUE TITLE
Stop creating tuple types per invocation

### ISSUE THREAD

**Issue by warunalakshitha** [ballerina-platform/ballerina-lang#32728]
**Description:**
```ballerina

type SingleStringParserTestCase [string, string, string, string];

public function main() {
    string s = "Waru";
        SingleStringParserTestCase[] sources =
        [[s, "1", s, "m"],
        ["q", "2", s, "n"],
        ["w", "3", s, "b"],
         ["w", "3", "s", "b"]];
}

```

Here once you check generated code, we have create tuple type for each array eleemnt.

**Comment by warunalakshitha** [comment]
Need same fix as https://github.com/ballerina-platform/ballerina-lang/issues/32019

**Comment by HindujaB** [comment]
Fixed with #32896
